app/models/user.rb

```diff
@@ -21,4 +26,8 @@ def self.find_or_create_from_auth_hash! auth_hash
   def owner_of? record
     record.user_id == self.id
   end
+
+  def role? name
+    roles.find { |role| role.name == name.to_s }
```

Comment:
Does ActiveRecord override and query with this proc? Or does roles return an array and you're iterating the array?

Reply:
Returns an active record association technically. I thought about making the `roles` an array column on the user and might still go that route.

Reply:
It's not uncommon to use a bitmask for that, but I find that you're eventually going to want to decorate the role assignment with some data... so I'd keep it in a separate table.